formal-spec/candidateproperties.tex

```diff
@@ -0,0 +1,153 @@
+\section{Candidate Properties}
+
+\subsection{Global Properties}
+
+\begin{property}[Software Update Proposal]\label{prop:submissions-for-all}
+Any stakeholder will always be able to propose a software update to be voted by the stakeholders' community and this will be recorded as an on-chain immutable event on the blockchain.
+\end{property}
+
+\begin{property}[Non Approved Software Updates]\label{prop:nonapproved-sus}
+A software update that is not approved by the stakeholders' community will never be applied,
+\end{property}
+
+\begin{property}[Approved Software Updates]\label{prop:approved-sus}
+A software update that has been approved by the stakeholders' community will be eventually applied.
```

Comment:
I'm not sure about this. What happens if the **implemented** update proposal is approved, but then nobody downloads the new version?

Also what happens if two proposals get approved at the same time? Did we consider this case in the paper? Otherwise we might need a property that states that at any time there is at most one proposal in the activation phase.